airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/global_parent_cursor.py

```diff
@@ -0,0 +1,197 @@
+            StreamSlice(partition=partition, cursor_slice=cursor_slice)
+            for partition in self._partition_router.stream_slices()
+            for cursor_slice in self._stream_cursor.stream_slices()
+        )
+
+        yield from flag_last(slices)
+
+    def set_initial_state(self, stream_state: StreamState) -> None:
+        """
+        Set the initial state for the cursors.
+
+        This method initializes the state for the global cursor using the provided stream state.
+        If a partition state is provided in the stream state, it will update the corresponding global cursor with this state.
+
+        Additionally, it sets the parent state for partition routers that are based on parent streams. If a partition router
+        does not have parent streams, this step will be skipped due to the default PartitionRouter implementation.
+
+        Args:
+            stream_state (StreamState): The state of the streams to be set. The format of the stream state should be:
+                {
+                    "state": {
+                        "last_updated": "2023-05-27T00:00:00Z"
+                    },
+                    "parent_state": {
+                        "parent_stream_name": {
+                            "last_updated": "2023-05-27T00:00:00Z"
+                        }
+                    }
+                }
+        """
+        if not stream_state:
+            return
+
+        self._stream_cursor.set_initial_state(stream_state["state"])
+
+        # Set parent state for partition routers based on parent streams
+        self._partition_router.set_initial_state(stream_state)
+
+    def observe(self, stream_slice: StreamSlice, record: Record) -> None:
+        self._stream_cursor.observe(StreamSlice(partition={}, cursor_slice=stream_slice.cursor_slice), record)
+
+    def close_slice(self, stream_slice: StreamSlice, *args: Any) -> None:
+        """
+        Close the current stream slice.
+
+        This method is called when a stream slice is completed. For the global parent cursor, we close the child cursor
+        only after reading all slices. This ensures that we do not miss any child records from a later parent record
+        if the child cursor is earlier than a record from the first parent record.
+
+        Args:
+            stream_slice (StreamSlice): The stream slice to be closed.
+            *args (Any): Additional arguments.
+        """
+        if stream_slice.last_slice:
+            self._stream_cursor.close_slice(StreamSlice(partition={}, cursor_slice=stream_slice.cursor_slice), *args)
+
+    def get_stream_state(self) -> StreamState:
+        state: dict[str, Any] = {"state": self._stream_cursor.get_stream_state()}
+
+        parent_state = self._partition_router.get_stream_state()
+        if parent_state:
+            state["parent_state"] = parent_state
+
+        return state
+
+    def select_state(self, stream_slice: Optional[StreamSlice] = None) -> Optional[StreamState]:
```

Comment:
let's add a comment mentioning the stream_slice param is ignored

Reply:
Added